util/dataset.go

```diff
@@ -84,6 +85,11 @@ type Os struct {
 	Windows interface{} `config:"windows" json:"windows,omitempty" yaml:"windows,omitempty"`
 }
 
+type Elasticsearch struct {
+	IndexTemplateSettings map[string]interface{} `config:"index-template.settings" json:"index-template.settings,omitempty" yaml:"index-template.settings,omitempty"`
```

Comment:
Nit: casing. For ingest pipelines we use `ingest_pipeline` (underscore-delimited) but for index templates we use `index-template` (hyphen-delimited).

Reply:
Glad you bring this up. I remember we had in the past a discussion that we should standardise. You have any preference?

Reply:
tl;dr: I lean towards underscore.

When it comes to defining these fields in JSON or YAML I don't think it matters much, especially since we would never start a field name with a delimiter (otherwise hyphens could be problematic in YAML). 

When it comes to deserializing these fields into language constructs, hyphen could be problematic in some languages as it might represent subtraction. 

So I lean towards underscore.

Reply:
Let's clean it up and go with underscore!

Reply:
I filed https://github.com/elastic/package-registry/issues/565 for this and will clean up the PR.